### ISSUE TITLE
Some questions about OT loss in RetroMAE v2

### ISSUE THREAD

**Issue by XChen-Zero** [staoxiao/RetroMAE#22]
It's an amazing job!
```py
    def ot_embedding(self, logits, attention_mask):
        mask = (1 - attention_mask.unsqueeze(-1)) * -1000
        reps, _ = torch.max(logits + mask, dim=1)  # B V
        return reps

    def decoder_ot_loss(self, ot_embedding, bag_word_weight):
        input = F.log_softmax(ot_embedding, dim=-1)
        bow_loss = torch.mean(-torch.sum(bag_word_weight * input, dim=1))
        return bow_loss
```
I have some questions, if the loss is meant to simulate BoW, can it be understood that each position in $V$ is the number of times that token appears?
If I understand correctly, is it OK to $sum(logits)$ instead of $max(logits)$?

**Comment by staoxiao** [comment]
> It's an amazing job!
> 
> ```python
>     def ot_embedding(self, logits, attention_mask):
>         mask = (1 - attention_mask.unsqueeze(-1)) * -1000
>         reps, _ = torch.max(logits + mask, dim=1)  # B V
>         return reps
> 
>     def decoder_ot_loss(self, ot_embedding, bag_word_weight):
>         input = F.log_softmax(ot_embedding, dim=-1)
>         bow_loss = torch.mean(-torch.sum(bag_word_weight * input, dim=1))
>         return bow_loss
> ```
> 
> I have some questions, if the loss is meant to simulate BoW, can it be understood that each position in V is the number of times that token appears? If I understand correctly, is it OK to sum(logits) instead of max(logits)?

Thanks for your interest in our work!
Each position in V is not the number of times that token appears. The most frequently appearing tokens are typically non-significant ones, such as "the", and "a". Therefore, using sum(logits) may enhance the impact of some unimportant tokens.

**Comment by staoxiao** [comment]
There are some experimental results in Splade v2: https://arxiv.org/pdf/2109.10086.pdf 
max operation is better than sum/mean operation.
Colbert also has the same obeservation.
